From 710a23b241c4e1e9020dff44cceb494b0d073990 Mon Sep 17 00:00:00 2001
From: Michael Natterer <mitch@gimp.org>
Date: Tue, 28 Aug 2012 12:54:28 +0200
Subject: [PATCH 03/16] gtk: support gtk-range-jump-to-click in GtkRange

---
 gtk/gtkrange.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/gtk/gtkrange.c b/gtk/gtkrange.c
index 5fcbef0..81ed8e1 100644
--- a/gtk/gtkrange.c
+++ b/gtk/gtkrange.c
@@ -2317,8 +2317,9 @@ gtk_range_button_press (GtkWidget      *widget,
 {
   GtkRange *range = GTK_RANGE (widget);
   gboolean primary_warps;
+  gboolean jump_to_click;
   gint page_increment_button, warp_button;
-  
+
   if (!gtk_widget_has_focus (widget))
     gtk_widget_grab_focus (widget);
 
@@ -2334,7 +2335,11 @@ gtk_range_button_press (GtkWidget      *widget,
   gtk_widget_style_get (widget,
                         "primary-button-warps-slider", &primary_warps,
                         NULL);
-  if (primary_warps)
+  g_object_get (gtk_widget_get_settings (widget),
+                "gtk-range-jump-to-click", &jump_to_click,
+                NULL);
+
+  if (primary_warps || jump_to_click)
     {
       warp_button = 1;
       page_increment_button = 3;
-- 
1.7.5.4

