From 3f7cc70439b7e9a55a02ff7f801394c681c597de Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 12 Feb 2013 14:04:37 +0100
Subject: [PATCH 35/36] entry: Use scaled icons on windows with a scale factor

---
 gtk/gtkentry.c |   36 ++++++++++++++++++++++--------------
 1 file changed, 22 insertions(+), 14 deletions(-)

diff --git a/gtk/gtkentry.c b/gtk/gtkentry.c
index 165840d..daaaa46 100644
--- a/gtk/gtkentry.c
+++ b/gtk/gtkentry.c
@@ -108,6 +108,7 @@ typedef struct
 
   GtkTargetList *target_list;
   GdkDragAction actions;
+  gdouble render_scale;
 } EntryIconInfo;
 
 struct _GtkEntryPrivate 
@@ -6511,15 +6512,17 @@ gtk_entry_ensure_pixbuf (GtkEntry             *entry,
     case GTK_IMAGE_STOCK:
       state = gtk_widget_get_state (widget);
       gtk_widget_set_state (widget, GTK_STATE_NORMAL);
-      icon_info->pixbuf = gtk_widget_render_icon (widget,
-                                                  icon_info->stock_id,
-                                                  GTK_ICON_SIZE_MENU,
-                                                  NULL);
+      icon_info->pixbuf = gtk_widget_render_icon_scaled (widget,
+                                                         icon_info->stock_id,
+                                                         GTK_ICON_SIZE_MENU,
+                                                         NULL,
+                                                         &icon_info->render_scale);
       if (!icon_info->pixbuf)
-        icon_info->pixbuf = gtk_widget_render_icon (widget,
-                                                    GTK_STOCK_MISSING_IMAGE,
-                                                    GTK_ICON_SIZE_MENU,
-                                                    NULL);
+        icon_info->pixbuf = gtk_widget_render_icon_scaled (widget,
+                                                           GTK_STOCK_MISSING_IMAGE,
+                                                           GTK_ICON_SIZE_MENU,
+                                                           NULL,
+                                                           &icon_info->render_scale);
       gtk_widget_set_state (widget, state);
       break;
 
@@ -6530,9 +6533,13 @@ gtk_entry_ensure_pixbuf (GtkEntry             *entry,
           icon_theme = gtk_icon_theme_get_for_screen (screen);
           settings = gtk_settings_get_for_screen (screen);
           
-          gtk_icon_size_lookup_for_settings (settings,
-                                             GTK_ICON_SIZE_MENU,
-                                             &width, &height);
+          gtk_icon_size_lookup_scaled (settings,
+                                       GTK_ICON_SIZE_MENU,
+                                       gdk_window_get_scale_factor (widget->window),
+                                       &width, &height);
+
+          g_print ("............... %f %d %d | %s\n", gdk_window_get_scale_factor (widget->window),
+                   width, height, icon_info->icon_name);
 
           icon_info->pixbuf = gtk_icon_theme_load_icon (icon_theme,
                                                         icon_info->icon_name,
@@ -6559,9 +6566,10 @@ gtk_entry_ensure_pixbuf (GtkEntry             *entry,
           icon_theme = gtk_icon_theme_get_for_screen (screen);
           settings = gtk_settings_get_for_screen (screen);
 
-          gtk_icon_size_lookup_for_settings (settings,
-                                             GTK_ICON_SIZE_MENU,
-                                             &width, &height);
+          gtk_icon_size_lookup_scaled (settings,
+                                       GTK_ICON_SIZE_MENU,
+                                       gdk_window_get_scale_factor (widget->window),
+                                       &width, &height);
 
           info = gtk_icon_theme_lookup_by_gicon (icon_theme,
                                                  icon_info->gicon,
-- 
1.7.10.2 (Apple Git-33)

