--- a/gtk/gtkscrolledwindow.c	2012-12-15 10:11:21.000000000 +0100
+++ b/gtk/gtkscrolledwindow.c	2012-12-15 11:22:02.000000000 +0100
@@ -3249,6 +3249,12 @@ gtk_scrolled_window_unrealize (GtkWidget
   GtkScrolledWindow *scrolled_window = GTK_SCROLLED_WINDOW (widget);
   GtkScrolledWindowPrivate *priv = GTK_SCROLLED_WINDOW_GET_PRIVATE (scrolled_window);
 
+  if (priv->sb_pointer_grabbed)
+    {
+      gtk_grab_remove (widget);
+      priv->sb_pointer_grabbed = FALSE;
+    }
+
   gdk_window_set_user_data (priv->overshoot_window, NULL);
   gdk_window_destroy (priv->overshoot_window);
   priv->overshoot_window = NULL;
