diff --git a/build/m4/banshee/osx.m4 b/build/m4/banshee/osx.m4
index b41ae07..1397070 100644
--- a/build/m4/banshee/osx.m4
+++ b/build/m4/banshee/osx.m4
@@ -1,6 +1,6 @@
 AC_DEFUN([BANSHEE_CHECK_OSX],
 [
-	IGEMACINTEGRATIONSHARP_REQUIRED=0.8.2
+	IGEMACINTEGRATIONSHARP_REQUIRED=0.8.6
 
 	AC_ARG_ENABLE(osx, AC_HELP_STRING([--enable-osx], [Enable OSX support]), enable_osx=$enableval, enable_osx="no")
 
@@ -9,11 +9,11 @@ AC_DEFUN([BANSHEE_CHECK_OSX],
 		have_osx="yes"
 
 		PKG_CHECK_MODULES(IGEMACINTEGRATIONSHARP, 
-			ige-mac-integration-sharp >= $IGEMACINTEGRATIONSHARP_REQUIRED,
+			ige-mac-integration >= $IGEMACINTEGRATIONSHARP_REQUIRED,
 			have_igemacintegrationsharp=yes, have_igemacintegrationsharp=no)
 	
 		if test "x$have_igemacintegrationsharp" = "xno"; then
-			AC_MSG_ERROR([ige-mac-integration-sharp was not found or is not up to date. Please install ige-mac-integration-sharp of at least version $IGEMACINTEGRATIONSHARP_REQUIRED])
+			AC_MSG_ERROR([ige-mac-integration was not found or is not up to date. Please install ige-mac-integration-sharp of at least version $IGEMACINTEGRATIONSHARP_REQUIRED])
 		fi
 		AC_SUBST(IGEMACINTEGRATIONSHARP_LIBS)
 	fi
diff --git a/configure b/configure
index 182c0f0..c11b6a0 100755
--- a/configure
+++ b/configure
@@ -17658,7 +17658,7 @@ fi
 
 
 
-	IGEMACINTEGRATIONSHARP_REQUIRED=0.8.2
+	IGEMACINTEGRATIONSHARP_REQUIRED=0.8.6
 
 	# Check whether --enable-osx was given.
 if test "${enable_osx+set}" = set; then
@@ -17680,12 +17680,12 @@ if test -n "$IGEMACINTEGRATIONSHARP_CFLAGS"; then
     pkg_cv_IGEMACINTEGRATIONSHARP_CFLAGS="$IGEMACINTEGRATIONSHARP_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"ige-mac-integration-sharp >= \$IGEMACINTEGRATIONSHARP_REQUIRED\"") >&5
-  ($PKG_CONFIG --exists --print-errors "ige-mac-integration-sharp >= $IGEMACINTEGRATIONSHARP_REQUIRED") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"ige-mac-integration >= \$IGEMACINTEGRATIONSHARP_REQUIRED\"") >&5
+  ($PKG_CONFIG --exists --print-errors "ige-mac-integration >= $IGEMACINTEGRATIONSHARP_REQUIRED") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_IGEMACINTEGRATIONSHARP_CFLAGS=`$PKG_CONFIG --cflags "ige-mac-integration-sharp >= $IGEMACINTEGRATIONSHARP_REQUIRED" 2>/dev/null`
+  pkg_cv_IGEMACINTEGRATIONSHARP_CFLAGS=`$PKG_CONFIG --cflags "ige-mac-integration >= $IGEMACINTEGRATIONSHARP_REQUIRED" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -17696,12 +17696,12 @@ if test -n "$IGEMACINTEGRATIONSHARP_LIBS"; then
     pkg_cv_IGEMACINTEGRATIONSHARP_LIBS="$IGEMACINTEGRATIONSHARP_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"ige-mac-integration-sharp >= \$IGEMACINTEGRATIONSHARP_REQUIRED\"") >&5
-  ($PKG_CONFIG --exists --print-errors "ige-mac-integration-sharp >= $IGEMACINTEGRATIONSHARP_REQUIRED") 2>&5
+    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"ige-mac-integration >= \$IGEMACINTEGRATIONSHARP_REQUIRED\"") >&5
+  ($PKG_CONFIG --exists --print-errors "ige-mac-integration >= $IGEMACINTEGRATIONSHARP_REQUIRED") 2>&5
   ac_status=$?
   $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_IGEMACINTEGRATIONSHARP_LIBS=`$PKG_CONFIG --libs "ige-mac-integration-sharp >= $IGEMACINTEGRATIONSHARP_REQUIRED" 2>/dev/null`
+  pkg_cv_IGEMACINTEGRATIONSHARP_LIBS=`$PKG_CONFIG --libs "ige-mac-integration >= $IGEMACINTEGRATIONSHARP_REQUIRED" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -17719,9 +17719,9 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        IGEMACINTEGRATIONSHARP_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "ige-mac-integration-sharp >= $IGEMACINTEGRATIONSHARP_REQUIRED" 2>&1`
+	        IGEMACINTEGRATIONSHARP_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "ige-mac-integration >= $IGEMACINTEGRATIONSHARP_REQUIRED" 2>&1`
         else
-	        IGEMACINTEGRATIONSHARP_PKG_ERRORS=`$PKG_CONFIG --print-errors "ige-mac-integration-sharp >= $IGEMACINTEGRATIONSHARP_REQUIRED" 2>&1`
+	        IGEMACINTEGRATIONSHARP_PKG_ERRORS=`$PKG_CONFIG --print-errors "ige-mac-integration >= $IGEMACINTEGRATIONSHARP_REQUIRED" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$IGEMACINTEGRATIONSHARP_PKG_ERRORS" >&5
@@ -17740,8 +17740,8 @@ $as_echo "yes" >&6; }
 fi
 
 		if test "x$have_igemacintegrationsharp" = "xno"; then
-			{ { $as_echo "$as_me:$LINENO: error: ige-mac-integration-sharp was not found or is not up to date. Please install ige-mac-integration-sharp of at least version $IGEMACINTEGRATIONSHARP_REQUIRED" >&5
-$as_echo "$as_me: error: ige-mac-integration-sharp was not found or is not up to date. Please install ige-mac-integration-sharp of at least version $IGEMACINTEGRATIONSHARP_REQUIRED" >&2;}
+			{ { $as_echo "$as_me:$LINENO: error: ige-mac-integration was not found or is not up to date. Please install ige-mac-integration of at least version $IGEMACINTEGRATIONSHARP_REQUIRED" >&5
+$as_echo "$as_me: error: ige-mac-integration was not found or is not up to date. Please install ige-mac-integration of at least version $IGEMACINTEGRATIONSHARP_REQUIRED" >&2;}
    { (exit 1); exit 1; }; }
 		fi
 
