diff --git a/pango/pangocoretext-fontmap.c b/pango/pangocoretext-fontmap.c
index 687c340..3ebc36d 100644
--- a/pango/pangocoretext-fontmap.c
+++ b/pango/pangocoretext-fontmap.c
@@ -1314,6 +1314,9 @@ pango_core_text_font_map_init (PangoCoreTextFontMap *ctfontmap)
       CTFontDescriptorRef desc = CFArrayGetValueAtIndex (ctfaces, i);
 
       str = CTFontDescriptorCopyAttribute (desc, kCTFontFamilyNameAttribute);
+      if (!str)
+        continue;
+
       buffer = gchar_from_cf_string (str);
 
       family_name = g_utf8_casefold (buffer, -1);
