
--- a/gettext-tools/libgettextpo/string.in.h	Sun Apr 25 02:22:40 2010
+++ b/gettext-tools/libgettextpo/string.in.h	Tue Oct  4 14:36:31 2011
@@ -229,6 +229,7 @@
 #if @GNULIB_STPNCPY@
 # if @REPLACE_STPNCPY@
 #  if !(defined __cplusplus && defined GNULIB_NAMESPACE)
+#   undef stpncpy
 #   define stpncpy rpl_stpncpy
 #  endif
 _GL_FUNCDECL_RPL (stpncpy, char *,
 
 	Rename module 'memxfrm' to 'amemxfrm'.
--- a/gettext-tools/gnulib-lib/string.in.h	Mon May 24 02:42:47 2010
+++ b/gettext-tools/gnulib-lib/string.in.h	Tue Oct  4 14:35:46 2011
@@ -229,6 +229,7 @@
 #if @GNULIB_STPNCPY@
 # if @REPLACE_STPNCPY@
 #  if !(defined __cplusplus && defined GNULIB_NAMESPACE)
+#   undef stpncpy
 #   define stpncpy rpl_stpncpy
 #  endif
 _GL_FUNCDECL_RPL (stpncpy, char *,

--- a/gettext-runtime/gnulib-lib/string.in.h	Sun Apr 25 02:20:42 2010
+++ b/gettext-runtime/gnulib-lib/string.in.h	Tue Oct  4 14:34:57 2011
@@ -229,6 +229,7 @@
 #if @GNULIB_STPNCPY@
 # if @REPLACE_STPNCPY@
 #  if !(defined __cplusplus && defined GNULIB_NAMESPACE)
+#   undef stpncpy
 #   define stpncpy rpl_stpncpy
 #  endif
 _GL_FUNCDECL_RPL (stpncpy, char *,
--- a/gettext-tools/configure	Sun Jun  6 13:12:20 2010
+++ b/gettext-tools/configure	Tue Oct  4 16:29:27 2011
@@ -40562,6 +40562,16 @@
 
 
 
+      ac_fn_c_check_decl "$LINENO" "stpncpy" "ac_cv_have_decl_stpncpy" "$ac_includes_default"
+if test "x$ac_cv_have_decl_stpncpy" = x""yes; then :
+  ac_have_decl=1
+else
+  ac_have_decl=0
+fi
+
+cat >>confdefs.h <<_ACEOF
+#define HAVE_DECL_STPNCPY $ac_have_decl
+_ACEOF
 
 
 
@@ -40606,7 +40616,9 @@
 #include <stdlib.h>
 #include <string.h> /* for strcpy */
 /* The stpncpy prototype is missing in <string.h> on AIX 4.  */
+#if !HAVE_DECL_STPNCPY
 extern char *stpncpy (char *dest, const char *src, size_t n);
+#endif
 int main () {
   const char *src = "Hello";
   char dest[10];
