From 45b5375e5579a189daca64489edab35c7a60066f Mon Sep 17 00:00:00 2001
From: Aaron Bockover <aaron@abock.org>
Date: Fri, 25 Feb 2011 13:32:26 -0500
Subject: [PATCH] Fix linking issues with latest pixman

---
 Makefile.am  |    4 ++--
 configure.ac |    4 +++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index cd7162d..1546cc3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-INCLUDES = $(GTK_CFLAGS)
+INCLUDES = $(GTK_CFLAGS) $(PIXMAN_CFLAGS)
 
 SUBDIRS = schema
 DIST_SUBDIRS = $(SUBDIRS) po
@@ -35,5 +35,5 @@ libmurrine_la_SOURCES = \
 	./src/stack-blur.h
 
 libmurrine_la_LDFLAGS = -module -avoid-version -no-undefined
-libmurrine_la_LIBADD =  $(GTK_LIBS)
+libmurrine_la_LIBADD =  $(GTK_LIBS) $(PIXMAN_LIBS)
 
diff --git a/configure.ac b/configure.ac
index 175d394..b3e5f9e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,10 +42,12 @@ AC_ARG_ENABLE(rgba,
 PKG_CHECK_MODULES(GTK, gtk+-2.0 >= 2.18.0,,
                   AC_MSG_ERROR([GTK+-2.18 is required to compile murrine]))
 
-PKG_CHECK_MODULES(pixman,[pixman-1],,AC_MSG_ERROR([Pixman is required to compile murrine]))
+PKG_CHECK_MODULES(PIXMAN,[pixman-1],,AC_MSG_ERROR([Pixman is required to compile murrine]))
 
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
+AC_SUBST(PIXMAN_CFLAGS)
+AC_SUBST(PIXMAN_LIBS)
 
 GTK_VERSION=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
 AC_SUBST(GTK_VERSION)
-- 
1.7.4

