diff --git a/pango/pangocairo-coretextfontmap.c b/pango/pangocairo-coretextfontmap.c
index 6cb3809..443694b 100644
--- a/pango/pangocairo-coretextfontmap.c
+++ b/pango/pangocairo-coretextfontmap.c
@@ -186,8 +186,9 @@ pango_cairo_core_text_font_map_load_font (PangoCoreTextFontMap       *fontmap,
                                           const PangoFontDescription *description)
 {
   char *descname = pango_font_description_get_family (description);
+  NSOperatingSystemVersion version = [[NSProcessInfo processInfo] operatingSystemVersion];
 
-  if (strcmp (descname, "-apple-system-font") == 0)
+  if (version.majorVersion == 10 && version.minorVersion >= 11 &&  strcmp (descname, "-apple-system-font") == 0)
     {
       PangoCoreTextFont *cfont;
       NSFont *sysfont;
diff --git a/pango/pangocoretext-fontmap.c b/pango/pangocoretext-fontmap.c
index 4234bf1..e7337c2 100644
--- a/pango/pangocoretext-fontmap.c
+++ b/pango/pangocoretext-fontmap.c
@@ -1367,6 +1367,7 @@ pango_core_text_font_map_init (PangoCoreTextFontMap *ctfontmap)
   CTFontCollectionRef collection;
   CFArrayRef ctfaces;
   CFIndex i, count;
+  NSOperatingSystemVersion version = [[NSProcessInfo processInfo] operatingSystemVersion];
 
   ctfontmap->serial = 1;
   ctfontmap->families = g_hash_table_new_full (g_str_hash, g_str_equal,
@@ -1430,7 +1431,7 @@ pango_core_text_font_map_init (PangoCoreTextFontMap *ctfontmap)
       CFRelease (dict);
     }
 
-  if (true)
+  if (version.majorVersion == 10 && version.minorVersion >= 11)
     {
       NSFont *sysfont = [NSFont systemFontOfSize: 0.0];
       NSString *name = [[NSFont fontWithName: [[sysfont fontDescriptor] postscriptName] size: 0.0] familyName];
