diff -r -u pango-1.29.5-old/configure pango-1.29.5/configure
--- pango-1.29.5-old/configure	2011-11-22 14:07:55.000000000 -0500
+++ pango-1.29.5/configure	2011-12-15 22:09:47.000000000 -0500
@@ -19508,7 +19508,7 @@
 #
 # Checks for GLib
 #
-GLIB_REQUIRED_VERSION=2.31.0
+GLIB_REQUIRED_VERSION=2.26.0
 GLIB_MODULES="glib-2.0 >= $GLIB_REQUIRED_VERSION gobject-2.0 gmodule-no-export-2.0"
 
 
diff -r -u pango-1.29.5-old/pango/pango-bidi-type.h pango-1.29.5/pango/pango-bidi-type.h
--- pango-1.29.5-old/pango/pango-bidi-type.h	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/pango-bidi-type.h	2011-12-15 22:09:26.000000000 -0500
@@ -126,7 +126,6 @@
 					     gint         length);
 
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(g_unichar_get_mirror_char)
 gboolean       pango_get_mirror_char        (gunichar     ch,
 					     gunichar    *mirrored_ch);
 #endif
diff -r -u pango-1.29.5-old/pango/pangocairo.h pango-1.29.5/pango/pangocairo.h
--- pango-1.29.5-old/pango/pangocairo.h	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/pangocairo.h	2011-12-15 22:09:26.000000000 -0500
@@ -75,7 +75,6 @@
 						   double             dpi);
 double        pango_cairo_font_map_get_resolution (PangoCairoFontMap *fontmap);
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(pango_font_map_create_context)
 PangoContext *pango_cairo_font_map_create_context (PangoCairoFontMap *fontmap);
 #endif
 
diff -r -u pango-1.29.5-old/pango/pangofc-font.h pango-1.29.5/pango/pangofc-font.h
--- pango-1.29.5-old/pango/pangofc-font.h	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/pangofc-font.h	2011-12-15 22:09:26.000000000 -0500
@@ -134,7 +134,6 @@
 guint      pango_fc_font_get_glyph         (PangoFcFont      *font,
 					    gunichar          wc);
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(PANGO_GET_UNKNOWN_GLYPH)
 PangoGlyph pango_fc_font_get_unknown_glyph (PangoFcFont      *font,
 					    gunichar          wc);
 #endif /* PANGO_DISABLE_DEPRECATED */
diff -r -u pango-1.29.5-old/pango/pangofc-fontmap.h pango-1.29.5/pango/pangofc-fontmap.h
--- pango-1.29.5-old/pango/pangofc-fontmap.h	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/pangofc-fontmap.h	2011-12-15 22:09:26.000000000 -0500
@@ -186,7 +186,6 @@
 };
 
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(pango_font_map_create_context)
 PangoContext * pango_fc_font_map_create_context (PangoFcFontMap *fcfontmap);
 #endif
 void           pango_fc_font_map_shutdown       (PangoFcFontMap *fcfontmap);
diff -r -u pango-1.29.5-old/pango/pangoft2.h pango-1.29.5/pango/pangoft2.h
--- pango-1.29.5-old/pango/pangoft2.h	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/pangoft2.h	2011-12-15 22:09:26.000000000 -0500
@@ -86,7 +86,6 @@
 							 GDestroyNotify          notify);
 void          pango_ft2_font_map_substitute_changed     (PangoFT2FontMap         *fontmap);
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(pango_font_map_create_context)
 PangoContext *pango_ft2_font_map_create_context         (PangoFT2FontMap         *fontmap);
 #endif
 
@@ -94,23 +93,16 @@
 /* API for rendering modules
  */
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(pango_font_map_create_context)
 PangoContext      *pango_ft2_get_context          (double dpi_x,
 						   double dpi_y);
-G_DEPRECATED_FOR(pango_ft2_font_map_new)
 PangoFontMap      *pango_ft2_font_map_for_display (void);
-G_DEPRECATED
 void               pango_ft2_shutdown_display     (void);
 
-G_DEPRECATED_FOR(PANGO_GET_UNKNOWN_GLYPH)
 PangoGlyph     pango_ft2_get_unknown_glyph (PangoFont       *font);
-G_DEPRECATED_FOR(pango_fc_font_kern_glyphs)
 int            pango_ft2_font_get_kerning  (PangoFont       *font,
 					    PangoGlyph       left,
 					    PangoGlyph       right);
-G_DEPRECATED_FOR(pango_fc_font_lock_face)
 FT_Face        pango_ft2_font_get_face     (PangoFont       *font);
-G_DEPRECATED_FOR(pango_font_get_coverage)
 PangoCoverage *pango_ft2_font_get_coverage (PangoFont       *font,
 					    PangoLanguage   *language);
 #endif /* PANGO_DISABLE_DEPRECATED */
diff -r -u pango-1.29.5-old/pango/pangowin32.h pango-1.29.5/pango/pangowin32.h
--- pango-1.29.5-old/pango/pangowin32.h	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/pangowin32.h	2011-12-15 22:09:26.000000000 -0500
@@ -42,7 +42,6 @@
 /* Calls for applications
  */
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(pango_font_map_create_context)
 PangoContext * pango_win32_get_context        (void);
 #endif
 
@@ -73,7 +72,6 @@
  */
 
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(PANGO_GET_UNKNOWN_GLYPH)
 PangoGlyph     pango_win32_get_unknown_glyph  (PangoFont        *font,
 					       gunichar          wc);
 #endif /* PANGO_DISABLE_DEPRECATED */
diff -r -u pango-1.29.5-old/pango/pangox.h pango-1.29.5/pango/pangox.h
--- pango-1.29.5-old/pango/pangox.h	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/pangox.h	2011-12-15 22:09:26.000000000 -0500
@@ -39,17 +39,13 @@
 
 /* Calls for applications
  */
-G_DEPRECATED
 PangoContext * pango_x_get_context        (Display          *display);
-G_DEPRECATED
 void           pango_x_context_set_funcs  (PangoContext     *context,
 					   PangoGetGCFunc    get_gc_func,
 					   PangoFreeGCFunc   free_gc_func);
 
-G_DEPRECATED
 PangoFont *    pango_x_load_font          (Display          *display,
 					   const gchar      *spec);
-G_DEPRECATED
 void           pango_x_render             (Display          *display,
 					   Drawable          d,
 					   GC                gc,
@@ -57,14 +53,12 @@
 					   PangoGlyphString *glyphs,
 					   gint              x,
 					   gint              y);
-G_DEPRECATED
 void           pango_x_render_layout_line (Display          *display,
 					   Drawable          drawable,
 					   GC                gc,
 					   PangoLayoutLine  *line,
 					   int               x,
 					   int               y);
-G_DEPRECATED
 void           pango_x_render_layout      (Display          *display,
 					   Drawable          drawable,
 					   GC                gc,
@@ -80,20 +74,16 @@
 #define PANGO_X_GLYPH_SUBFONT(glyph) ((glyph)>>16)
 #define PANGO_X_GLYPH_INDEX(glyph) ((glyph) & 0xffff)
 
-G_DEPRECATED
 int        pango_x_list_subfonts     (PangoFont      *font,
 				      char          **charsets,
 				      int             n_charsets,
 				      PangoXSubfont **subfont_ids,
 				      int           **subfont_charsets);
-G_DEPRECATED
 gboolean   pango_x_has_glyph         (PangoFont      *font,
 				      PangoGlyph      glyph);
-G_DEPRECATED
 PangoGlyph pango_x_get_unknown_glyph (PangoFont      *font);
 
 #ifdef PANGO_ENABLE_ENGINE
-G_DEPRECATED
 PangoGlyph pango_x_font_get_unknown_glyph (PangoFont    *font,
 					   gunichar      wc);
 #endif /* PANGO_ENABLE_ENGINE */
@@ -102,43 +92,32 @@
  */
 typedef struct _PangoXFontCache PangoXFontCache;
 
-G_DEPRECATED
 PangoXFontCache *pango_x_font_cache_new     (Display         *display);
-G_DEPRECATED
 void             pango_x_font_cache_free    (PangoXFontCache *cache);
 
-G_DEPRECATED
 XFontStruct * pango_x_font_cache_load      (PangoXFontCache *cache,
 					    const char      *xlfd);
-G_DEPRECATED
 void          pango_x_font_cache_unload    (PangoXFontCache *cache,
 					    XFontStruct     *fs);
 
-G_DEPRECATED
 PangoFontMap *   pango_x_font_map_for_display  (Display     *display);
-G_DEPRECATED
 void             pango_x_shutdown_display      (Display     *display);
-G_DEPRECATED
 PangoXFontCache *pango_x_font_map_get_font_cache (PangoFontMap *font_map);
 
-G_DEPRECATED
 char *pango_x_font_subfont_xlfd (PangoFont     *font,
 				 PangoXSubfont  subfont_id);
 
 
-G_DEPRECATED
 gboolean pango_x_find_first_subfont (PangoFont     *font,
 				     char         **charsets,
 				     int            n_charsets,
 				     PangoXSubfont *rfont);
 
-G_DEPRECATED
 void pango_x_fallback_shape (PangoFont        *font,
 			     PangoGlyphString *glyphs,
 			     const char       *text,
 			     int               n_chars);
 
-G_DEPRECATED
 gboolean pango_x_apply_ligatures (PangoFont     *font,
 				  PangoXSubfont  subfont,
 				  gunichar     **glyphs,
diff -r -u pango-1.29.5-old/pango/pangoxft.h pango-1.29.5/pango/pangoxft.h
--- pango-1.29.5-old/pango/pangoxft.h	2011-11-22 13:59:54.000000000 -0500
+++ pango-1.29.5/pango/pangoxft.h	2011-12-15 22:09:26.000000000 -0500
@@ -80,17 +80,12 @@
 XftFont *     pango_xft_font_get_font          (PangoFont *font);
 Display *     pango_xft_font_get_display       (PangoFont *font);
 #ifndef PANGO_DISABLE_DEPRECATED
-G_DEPRECATED_FOR(pango_fc_font_lock_face)
 FT_Face       pango_xft_font_lock_face         (PangoFont *font);
-G_DEPRECATED_FOR(pango_fc_font_unlock_face)
 void	      pango_xft_font_unlock_face       (PangoFont *font);
-G_DEPRECATED_FOR(pango_fc_font_get_glyph)
 guint	      pango_xft_font_get_glyph	       (PangoFont *font,
 						gunichar   wc);
-G_DEPRECATED_FOR(pango_fc_font_has_char)
 gboolean      pango_xft_font_has_char          (PangoFont *font,
 						gunichar   wc);
-G_DEPRECATED_FOR(PANGO_GET_UNKNOWN_GLYPH)
 PangoGlyph    pango_xft_font_get_unknown_glyph (PangoFont *font,
 						gunichar   wc);
 #endif /* PANGO_DISABLE_DEPRECATED */
