diff --git a/pango/pangocairo-coretextfontmap.c b/pango/pangocairo-coretextfontmap.c
index 4ad23fa..15fc832 100644
--- a/pango/pangocairo-coretextfontmap.c
+++ b/pango/pangocairo-coretextfontmap.c
@@ -246,6 +246,10 @@ pango_cairo_core_text_font_map_load_font (PangoCoreTextFontMap       *fontmap,
                                             pango_core_text_font_key_get_matrix (&key),
                                             &font_matrix);
 
+      g_object_ref (cfont);
+      return cfont;
+
+      /*
       PangoCoreTextFont *tmp_font = _pango_core_text_font_map_lookup_font (fontmap, &key);
       if (tmp_font)
         {
@@ -259,6 +263,7 @@ pango_cairo_core_text_font_map_load_font (PangoCoreTextFontMap       *fontmap,
 
           return cfont;
         }
+        */
     }
 
   return PANGO_FONT_MAP_CLASS (pango_cairo_core_text_font_map_parent_class)->load_font (fontmap, context, description);
