diff --git a/gdk/quartz/gdkwindow-quartz.c b/gdk/quartz/gdkwindow-quartz.c
index 0bb5449..97f2e90 100644
--- a/gdk/quartz/gdkwindow-quartz.c
+++ b/gdk/quartz/gdkwindow-quartz.c
@@ -3194,6 +3194,11 @@ _gdk_windowing_window_get_input_shape (GdkWindow *window)
   return NULL;
 }
 
+/* Protocol to build cleanly for OSX < 10.7 */
+@protocol ScaleFactor
+- (CGFloat) backingScaleFactor;
+@end
+
 static gdouble
 gdk_window_quartz_get_scale_factor (GdkWindow *window)
 {
@@ -3204,8 +3209,8 @@ gdk_window_quartz_get_scale_factor (GdkWindow *window)
 
   impl = GDK_WINDOW_IMPL_QUARTZ (GDK_WINDOW_OBJECT (window)->impl);
 
-  if ([impl->toplevel respondsToSelector:@selector(backingScaleFactor)])
-    return [impl->toplevel backingScaleFactor];
+  if (gdk_quartz_osx_version() >= GDK_OSX_LION)
+    return [(id <ScaleFactor>) impl->toplevel backingScaleFactor];
 
   return 1.0;
 }
