From abcb5dbad4381dcaba8f84043bc6a8deb2e1e2f6 Mon Sep 17 00:00:00 2001
From: Michael Natterer <mitch@gimp.org>
Date: Fri, 23 Mar 2012 16:27:54 +0100
Subject: [PATCH 4/5] tests: add a GtkEntry to testnsview so we can test focus
 better

---
 tests/testnsview.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/tests/testnsview.c b/tests/testnsview.c
index 5842529..85bad7c 100644
--- a/tests/testnsview.c
+++ b/tests/testnsview.c
@@ -152,6 +152,16 @@ main (gint   argc,
     [text_field release];
   }
 
+  /* and a normal GtkEntry to check focus */
+  {
+    GtkWidget *entry;
+
+    entry = gtk_entry_new ();
+    gtk_entry_set_text (GTK_ENTRY (entry), "Normal GTK+ entry");
+    gtk_box_pack_start (GTK_BOX (vbox), entry, FALSE, FALSE, 0);
+    gtk_widget_show (entry);
+  }
+
   gtk_widget_show (window);
 
   gtk_main ();
-- 
1.7.5.4

