diff --git a/Monobjc.build b/Monobjc.build
index 03546ab..83c350b 100644
--- a/Monobjc.build
+++ b/Monobjc.build
@@ -60,12 +60,16 @@
   ================================================================================ -->
   <target name="build-libraries" description="Build the two set of libraries" depends="banner, prepare">
     <!-- Native libraries -->
+    <!--
     <call target="build-libraries-native"/>
+    -->
     
     <!-- Version : 1.0 -->
+    <!--
     <property name="version" value="1.0"/>
     <property name="define" value="ENABLE_SHORT_KEY;ENABLE_FAST_PATH"/>
     <call target="build-libraries-intern"/>
+    -->
     
     <!-- Version : 2.0 -->
     <property name="version" value="2.0"/>
@@ -513,6 +517,7 @@ public static String FormatVersion(String version) {
     		output="${output.dir}/${lib.name}.dll"
     		doc="${output.dir}/${lib.name}.xml"
     		define="${define}"
+    		debug="true"
     		keyfile="${keyfile}">
       <sources>
         <include name="${build.dir}/GeneratedAssemblyInfo.${version}.${hg.revision}.0.cs"/>
