diff --git a/src/libraries/Monobjc/Runtime/ObjectiveC20/NativeMethods.cs b/src/libraries/Monobjc/Runtime/ObjectiveC20/NativeMethods.cs
index 74ce891..c44aa30 100644
--- a/src/libraries/Monobjc/Runtime/ObjectiveC20/NativeMethods.cs
+++ b/src/libraries/Monobjc/Runtime/ObjectiveC20/NativeMethods.cs
@@ -197,7 +197,7 @@ namespace Monobjc.Runtime.ObjectiveC20
         /// <summary>
         /// <para></para>
         /// </summary>
-        [DllImport("@executable_path/libmonobjc.2.dylib", EntryPoint = "hook_thread_lifecycle")]
+        [DllImport("libmonobjc.2.dylib", EntryPoint = "hook_thread_lifecycle")]
         public static extern int hook_thread_lifecycle();
 
         /// <summary>
@@ -205,7 +205,7 @@ namespace Monobjc.Runtime.ObjectiveC20
         /// </summary>
         /// <param name="callback"></param>
         /// <returns></returns>
-        [DllImport("@executable_path/libmonobjc.2.dylib", EntryPoint = "set_dealloc_callback")]
+        [DllImport("libmonobjc.2.dylib", EntryPoint = "set_dealloc_callback")]
         public static extern bool set_dealloc_callback(RuntimeBridge.DeallocCallback callback);
 
         /// <summary>
@@ -213,7 +213,7 @@ namespace Monobjc.Runtime.ObjectiveC20
         /// </summary>
         /// <param name="cls"></param>
         /// <returns></returns>
-        [DllImport("@executable_path/libmonobjc.2.dylib", EntryPoint = "intercept_dealloc_for")]
+        [DllImport("libmonobjc.2.dylib", EntryPoint = "intercept_dealloc_for")]
         public static extern bool intercept_dealloc_for(IntPtr cls);
 
         /// <summary>
@@ -221,7 +221,7 @@ namespace Monobjc.Runtime.ObjectiveC20
         /// </summary>
         /// <param name="target"></param>
         /// <returns></returns>
-        [DllImport("@executable_path/libmonobjc.2.dylib", EntryPoint = "add_object")]
+        [DllImport("libmonobjc.2.dylib", EntryPoint = "add_object")]
         public static extern bool add_object(IntPtr target);
 
         /// <summary>
@@ -229,7 +229,7 @@ namespace Monobjc.Runtime.ObjectiveC20
         /// </summary>
         /// <param name="target"></param>
         /// <returns></returns>
-        [DllImport("@executable_path/libmonobjc.2.dylib", EntryPoint = "contains_object")]
+        [DllImport("libmonobjc.2.dylib", EntryPoint = "contains_object")]
         public static extern int contains_object(IntPtr target);
 
         /// <summary>
@@ -237,8 +237,8 @@ namespace Monobjc.Runtime.ObjectiveC20
         /// </summary>
         /// <param name="target"></param>
         /// <returns></returns>
-        [DllImport("@executable_path/libmonobjc.2.dylib", EntryPoint = "remove_object")]
+        [DllImport("libmonobjc.2.dylib", EntryPoint = "remove_object")]
         public static extern bool remove_object(IntPtr target);
 #endif
     }
-}
\ No newline at end of file
+}
