From e81b6971d85c7a782269454311b022ce14787486 Mon Sep 17 00:00:00 2001
From: Michael Natterer <mitch@lanedo.com>
Date: Fri, 28 Oct 2011 08:19:24 +0000
Subject: gtkrc.key.mac: add '"' missing from last commit

---
diff --git a/gtk/gtkrc.key.mac b/gtk/gtkrc.key.mac
index 631e0dd..980f3e4 100644
--- a/gtk/gtkrc.key.mac
+++ b/gtk/gtkrc.key.mac
@@ -115,7 +115,7 @@ binding "gtk-mac-emacs-like"
 
 class "GtkTextView" binding "gtk-mac-emacs-like"
 class "GtkLabel" binding "gtk-mac-emacs-like"
-class "GtkEntry" binding "gtk-mac-emacs-like
+class "GtkEntry" binding "gtk-mac-emacs-like"
 
 
 binding "gtk-mac-file-chooser"
--
cgit v0.9.0.2
