--- a/gtk/gtkimage.c	2013-06-20 20:27:09.000000000 -0400
+++ b/gtk/gtkimage.c	2013-06-20 20:46:22.000000000 -0400
@@ -1761,7 +1761,7 @@
 	}
       else if (!gtk_icon_size_lookup_scaled (settings,
                                              image->icon_size,
-                                             gdk_window_get_scale_factor (GTK_WIDGET (image)->window),
+                                             gtk_widget_get_scale_factor (GTK_WIDGET (image)),
                                              &width, &height))
 	{
 	  if (image->icon_size == -1)
--- a/gtk/gtkicontheme.c	2013-06-20 19:55:15.000000000 -0400
+++ b/gtk/gtkicontheme.c	2013-06-20 19:57:17.000000000 -0400
@@ -1420,10 +1420,13 @@
                                                   size, 2, flags);
   if (variant)
     {
-      g_object_set_data_full (G_OBJECT (retval->pixbuf),
-                              "gdk-pixbuf-2x-variant",
-                              g_object_ref (variant->pixbuf),
-                              (GDestroyNotify) g_object_unref);
+      if (variant->pixbuf && retval->pixbuf)
+        {
+          g_object_set_data_full (G_OBJECT (retval->pixbuf),
+                                  "gdk-pixbuf-2x-variant",
+                                  g_object_ref (variant->pixbuf),
+                                  (GDestroyNotify) g_object_unref);
+        }
       gtk_icon_info_free (variant);
     }
 
@@ -1528,10 +1531,13 @@
 
   if (variant)
     {
-      g_object_set_data_full (G_OBJECT (retval->pixbuf),
-                              "gdk-pixbuf-2x-variant",
-                              g_object_ref (variant->pixbuf),
-                              (GDestroyNotify) g_object_unref);
+      if (variant->pixbuf && retval->pixbuf)
+        {
+          g_object_set_data_full (G_OBJECT (retval->pixbuf),
+                                  "gdk-pixbuf-2x-variant",
+                                  g_object_ref (variant->pixbuf),
+                                  (GDestroyNotify) g_object_unref);
+        }
       gtk_icon_info_free (variant);
     }
 
