*** a/gdk/quartz/gdkwindow-quartz.c~	Mon Sep 28 09:40:34 2009
--- b/gdk/quartz/gdkwindow-quartz.c	Sun Oct 18 16:32:26 2009
***************
*** 2984,2989 ****
--- 2984,3002 ----
  }
  
  void
+ _gdk_quartz_input_window_destroy(GdkWindow *window)
+ {
+   /* FIXME: Implement */
+ }
+ 
+ void
+ _gdk_quartz_input_window_crossing(GdkWindow *window,
+ 				 gboolean   enter)
+ {
+   /* FIXME: Implement */
+ }
+ 
+ void
  _gdk_windowing_window_set_composited (GdkWindow *window, gboolean composited)
  {
  }
***************
*** 3035,3038 ****
--- 3048,3053 ----
    iface->queue_antiexpose = _gdk_quartz_window_queue_antiexpose;
    iface->queue_translation = _gdk_quartz_window_queue_translation;
    iface->destroy = _gdk_quartz_window_destroy;
+   iface->input_window_destroy = _gdk_quartz_input_window_destroy;
+   iface->input_window_crossing = _gdk_quartz_input_window_crossing;
  }


