diff -c /Users/john/Library/Gnucash-svn/src/gtk\+-2.20.1/gdk/quartz/gdkeventloop-quartz.c\~ /Users/john/Library/Gnucash-svn/src/gtk\+-2.20.1/gdk/quartz/gdkeventloop-quartz.c
--- a/gdk/quartz/gdkeventloop-quartz.c	Thu Apr  8 19:29:55 2010
+++ b/gdk/quartz/gdkeventloop-quartz.c	Sat Jan  1 16:26:30 2011
@@ -640,13 +640,14 @@
    * releasing NSAutoReleasePools not properly nested with CFRunLoop
    * runs seems to cause problems.
    */
-  if (current_loop_level == 0)
+
+  if (current_loop_level == 0 && g_main_depth() == 0)
     {
       if (autorelease_pool)
 	[autorelease_pool release];
       autorelease_pool = [[NSAutoreleasePool alloc] init];
     }
-  
+
   retval = (_gdk_event_queue_find_first (_gdk_display) != NULL ||
 	    _gdk_quartz_event_loop_check_pending ());
 

Diff finished.  Sun Jan  2 17:43:09 2011
