diff --git a/gdk/quartz/gdkgc-quartz.c b/gdk/quartz/gdkgc-quartz.c
index 21cc598..9b79be0 100644
--- a/gdk/quartz/gdkgc-quartz.c
+++ b/gdk/quartz/gdkgc-quartz.c
@@ -463,6 +463,10 @@ _gdk_quartz_gc_update_cg_context (GdkGC                      *gc,
       gdk_region_get_rectangles (_gdk_gc_get_clip_region (gc),
 				 &rects, &n_rects);
 
+      /* No rects, nothing to do */
+      if (n_rects == 0)
+        return;
+
       if (n_rects == 1)
 	cg_rects = &rect;
       else
