diff --git a/gtk/gtknsview.c b/gtk/gtknsview.c
index a4b4dd4dbe..0fc5055ed0 100644
--- a/gtk/gtknsview.c
+++ b/gtk/gtknsview.c
@@ -442,15 +442,30 @@ gtk_ns_view_replace_draw_insertion_point (void)
     }
 }
 
+gboolean
+does_accept_first_responder_recursively (NSView *view)
+{
+  if ([view acceptsFirstResponder]) {
+    g_print ("%s accepts? %d\n", object_getClassName (view), [view acceptsFirstResponder]);
+    return TRUE;
+  }
+
+  for (NSView *subview in [view subviews]) {
+    return does_accept_first_responder_recursively (subview);
+  }
+
+  return FALSE;
+}
+
 static void
 gtk_ns_view_constructed (GObject *object)
 {
   GtkNSView *ns_view = GTK_NS_VIEW (object);
+  gboolean can_focus = does_accept_first_responder_recursively (ns_view->priv->view);
 
   G_OBJECT_CLASS (gtk_ns_view_parent_class)->constructed (object);
 
-  gtk_widget_set_can_focus (GTK_WIDGET (ns_view),
-                            [ns_view->priv->view acceptsFirstResponder]);
+  gtk_widget_set_can_focus (GTK_WIDGET (ns_view), can_focus);
 
 #if DEBUG_FOCUS
   g_printerr ("%s can focus: %d\n",
@@ -819,8 +834,7 @@ gtk_ns_view_native_child_event (GdkWindow *window,
               if (hit &&
                   (hit == view ||
                    [hit ancestorSharedWithView: view] == view) &&
-                  ([hit acceptsFirstResponder] ||
-                   [view acceptsFirstResponder]))
+                   does_accept_first_responder_recursively (view))
                 {
 #if DEBUG_FOCUS
                   g_printerr ("grabbing focus on %s\n",
