--- a/gtk/gtkselection-quartz.c	2010-09-20 14:45:32.000000000 -0700
+++ b/gtk/gtkselection-quartz.c	2010-12-31 15:40:31.000000000 -0800
@@ -560,7 +560,9 @@
       
       if (selection_info->selection == selection &&
 	  selection_info->owner == owner)
-	{	
+	{
+	  GtkClipboard *clip = gtk_clipboard_get(selection_info->selection);
+	  gtk_clipboard_clear(clip);
 	  current_selections = g_list_remove_link (current_selections,
 						   tmp_list);
 	  g_list_free (tmp_list);
--- a/gtk/gtkdnd-quartz.c	2010-05-01 20:14:28.000000000 -0700
+++ b/gtk/gtkdnd-quartz.c	2010-12-31 15:07:59.000000000 -0800
@@ -148,7 +148,8 @@
 {
   guint target_info;
   GtkSelectionData selection_data;
-
+  g_return_if_fail(info->source_widget != NULL);
+  g_return_if_fail(info->target_list != NULL);
   selection_data.selection = GDK_NONE;
   selection_data.data = NULL;
   selection_data.length = -1;
@@ -171,6 +172,13 @@
     }
 }
 
+- (void)pasteboardChangedOwner: (NSPasteboard*)sender
+{
+    info->target_list = NULL;
+    info->widget = NULL;
+    info->source_widget = NULL;
+}
+
 - (id)initWithInfo:(GtkDragSourceInfo *)anInfo
 {
   self = [super init];
@@ -1124,13 +1132,27 @@
 {
   GtkDragSourceInfo *info;
   GdkDragContext *context;
-  NSWindow *nswindow;
+  NSWindow *nswindow = get_toplevel_nswindow (widget);
+  NSPoint point = {(double)event->motion.x, (double)event->motion.y};
+  NSEvent *nsevent = [NSEvent mouseEventWithType: NSLeftMouseDown
+		      location: point
+		      modifierFlags: 0
+		      timestamp: event->motion.time
+		      windowNumber: [nswindow windowNumber]
+		      context: [nswindow graphicsContext]
+		      eventNumber: 0
+		      clickCount: 1
+		      pressure: 0.0 ];
+  g_return_val_if_fail(nsevent != NULL, NULL);
 
   context = gdk_drag_begin (NULL, NULL);
+  g_return_val_if_fail( context != NULL, NULL);
   context->is_source = TRUE;
 
   info = gtk_drag_get_source_info (context, TRUE);
-  
+  info->nsevent = nsevent;
+  [info->nsevent retain];
+
   info->source_widget = g_object_ref (widget);
   info->widget = g_object_ref (widget);
   info->target_list = target_list;
@@ -1197,9 +1213,6 @@
 	  }
     }
 
-  nswindow = get_toplevel_nswindow (widget);
-  info->nsevent = [nswindow currentEvent];
-  [info->nsevent retain];
 
   /* drag will begin in an idle handler to avoid nested run loops */
 
--- a/gdk/quartz/gdkdnd-quartz.c	2010-05-01 20:14:28.000000000 -0700
+++ b/gdk/quartz/gdkdnd-quartz.c	2010-12-31 12:15:13.000000000 -0800
@@ -111,7 +111,7 @@
 gdk_drag_begin (GdkWindow     *window,
 		GList         *targets)
 {
-  g_assert (_gdk_quartz_drag_source_context == NULL);
+  g_return_val_if_fail (_gdk_quartz_drag_source_context == NULL, NULL);
   
   /* Create fake context */
   _gdk_quartz_drag_source_context = gdk_drag_context_new ();
